apps/calendar/test/unit/views/settings_test.js

```diff
@@ -344,8 +349,23 @@ suiteGroup('views/settings', function() {
       calledWith = arguments;
     };
 
+    assert.notOk(subject.syncProgress.getAttribute('data-l10n-id'));
+    assert.notOk(subject.syncProgress.classList.contains('syncing'));
+
     triggerEvent(subject.syncButton, 'click');
     assert.ok(calledWith);
+
+    controller.on('syncStart', function () {
+      assert.ok(subject.syncProgress.classList.contains('syncing'));
+      assert.equal(subject.syncProgress.getAttribute('data-l10n-id'),
+        'sync-progress-syncing');
```

Comment:
these assertions are not working the way you expect.. `controller.on('syncStart', fn)` will be triggered asynchronously, so even if the assertions fails they should only fail after the `test` is already complete..

the proper way to test this is to create 2 async tests like:

```js
suite('syncProgress', function() {
  var syncController = app.syncController;

  test('syncStart', function(done) {
    syncController.on('syncStart', function () {
      assert.ok(subject.syncProgress.classList.contains('syncing'));
      assert.equal(subject.syncProgress.getAttribute('data-l10n-id'),
      'sync-progress-syncing');
      done();
    });
    syncController.emit('syncStart');
  });

  test('syncComplete', function(done) {
    syncController.on('syncComplete', function () {
      assert.equal(subject.syncProgress.getAttribute('data-l10n-id'),
      'sync-progress-complete');
      assert.notOk(subject.syncProgress.classList.contains('syncing'));
      done();
    });
    syncController.emit('syncComplete');
  });
});
```

or if you are really want to test the `click` event, you should use the [`waitFor` pattern](https://github.com/jobara/gaia/blob/1086930/apps/calendar/test/unit/controllers/notifications_test.js#L93-L98)